src/net/sf/mpxj/primavera/PrimaveraReader.java

```diff
@@ -692,33 +863,39 @@ private void addTaskUDFValue(Task task, Row row)
                break;
             }
 
+            case BOOLEAN:
+            {
+               value = STATICTYPE_UDF_MAP.get(row.getString("udf_text"));
```

Comment:
I wonder if we can pull out the UDF handling into a separate "manager" class, maybe to replace or augment the existing `UserFieldCounters` class. That way everyone gets the default implementation, and you can provide something more suited to your use case when you instantiated the reader.